infrastructure/server/network_security_groups.tf

```diff
@@ -0,0 +1,121 @@
+resource "aws_security_group" "fam_app_sg" {
+    name = "fam_app_sg"
+    description = "FAM custom security group for application tier (lambdas)."
+    vpc_id = data.aws_vpc.selected.id
+    revoke_rules_on_delete = true
+
+    tags = {
+        Name = "fam_app_sg"
+        managed-by = "terraform"
+    }
+
+    ingress {
+        from_port = 0
+        to_port = 0
+        protocol = "-1"
+        cidr_blocks = ["10.10.32.0/20", "10.10.128.0/20"]
+        description = "Central VPC Traffic Inbound from Web subnets"
+    }
+
+    ingress {
+        from_port = 0
+        to_port = 0
+        protocol = "-1"
+        cidr_blocks = ["10.10.0.0/19", "10.10.96.0/19"]
+        description = "Central VPC Traffic Inbound from App subnets"
+    }
+
+    ingress {
+        from_port = 0
+        to_port = 0
+        protocol = "-1"
+        cidr_blocks = ["10.10.64.0/21", "10.10.72.0/21"]
+        description = "Central VPC Traffic Inbound from Mgmt subnets"
+    }
+
+    egress {
+        from_port = 0
+        to_port = 0
+        protocol = "-1"
+        cidr_blocks = ["0.0.0.0/0"]
+        description = "Allow All Outbound Traffic"
+    }
+
+}
+
+resource "aws_security_group" "fam_data_sg" {
+    name = "fam_data_sg"
+    description = "FAM custom security group for data tier."
+    vpc_id = data.aws_vpc.selected.id
+    revoke_rules_on_delete = true
+    tags = {
+        Name = "fam_data_sg"
+        managed-by = "terraform"
+    }
+}
+
+resource "aws_vpc_security_group_ingress_rule" "fam_data_sg_east_west" {
+  security_group_id = aws_security_group.fam_data_sg.id
+  referenced_security_group_id = aws_security_group.fam_data_sg.id
```

Comment:
Just to confirm, because I see these two id values are **the same** "**security_group_id**" vs "**referenced_security_group_id**", I am guessing Terraform using this "rule resource" syntax to referencing to itself to effectively means "**allow all traffic/ports, and all regions**" as well. (including west/east)?

Reply:
security_group_id is the parent security group that this rule belongs to.
referenced_security_group_id is the source security group's traffic (as the source) that this rule applies to.

Compare to the other rule for fam_data_sg that is referencing the fam_app_sg security group.

Reply:
And yes, for other comment in teams - I likely forgot to update the other environments (dev/test/prod) terragrunt files like I did for tools.